### PR TITLE
Apply add mode to add license and copyright

### PR DESCRIPTION
**Pull request by bjk7119**
## Description
- Print file lists that have both license and copyright
- Add license / copyright at the header of missing license / copyright file
- Add manual and automatic mode to add copyright / license

## Type of change
Please insert 'x' one of the type of change.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/fosslight_reuse/_help.py [hunk 1]
```diff
@@ -7,20 +7,26 @@
 _HELP_MESSAGE_REUSE = """
     Usage: fosslight_reuse [Mode] [option1] <arg1> [option2] <arg2>...
         ex) fosslight_reuse lint -p /home/test/ -f "notice/sample.py,src/init.py"
+            fosslight_reuse add -a -l MIT -c LGE
```
Comment (soimkim) [inline src/fosslight_reuse/_help.py:10]: -a ? <-- legacy option?

Comment (soimkim) [reply src/fosslight_reuse/_help.py:10]: -c LGE is working?

File: src/fosslight_reuse/_add.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+_result_log = {}
+logger = logging.getLogger(constant.LOGGER_NAME)
+
+
+def check_file_extension(file_list):
+    files_filtered = []
+
+    if file_list != "":
+        for file in file_list:
+            file_extension = os.path.splitext(file)[1]
```
Comment (soimkim) [inline src/fosslight_reuse/_add.py:29]: An exception is required when there is no file extension.
- ex. file name = "test"

File: src/fosslight_reuse/_add.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            if file_extension in EXTENSION_COMMENT_STYLE_MAP_LOWERCASE:
```
Comment (soimkim) [inline src/fosslight_reuse/_add.py:30]: lower() is needed.

File: src/fosslight_reuse/_add.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        file_list = list()
+        file_list.append(file)
+
+        unused_lic_list, usused_cop_list, error_occurred, project = reuse_for_files(path_to_find, file_list)
+
+    return missing_license_filtered, missing_copyright_filtered
+
+
+def convert_to_spdx_format(input_string):
+    input_string = input_string.replace(" ", "-")
```
Comment (soimkim) [inline src/fosslight_reuse/_add.py:57]: Make it work only if input string is not SPDX-ID.

**Review by soimkim — CHANGES_REQUESTED** [review]

File: src/fosslight_reuse/_help.py (hunk 1 above)
Comment (bjk7119) [reply src/fosslight_reuse/_help.py:10]: Yes, -a is not working now and this example isn't correct. So, I'll update this example at next commit.

File: src/fosslight_reuse/_add.py (hunk 2 above)
Comment (bjk7119) [reply src/fosslight_reuse/_add.py:29]: okay. I agree with you, so I'll add exception line. Thank you

File: src/fosslight_reuse/_add.py (hunk 3 above)
Comment (bjk7119) [reply src/fosslight_reuse/_add.py:30]: I'll modify to always compare with lower case.

File: src/fosslight_reuse/_add.py (hunk 4 above)
Comment (bjk7119) [reply src/fosslight_reuse/_add.py:57]: Okay I'll add code to check if the input string is SPDX-ID

Comment (soimkim) [reply src/fosslight_reuse/_add.py:61]: Let's add the function for checking SPDX next time...

**Review by soimkim — APPROVED** [review]

**Pull request merged**